### DISCUSSION TITLE
Syntax coloring confusion

### DISCUSSION THREAD

**Discussion by StephanEggermont** [hedyorg/hedy#1861] [Implementation discussions]
My youngest just finished level 1. I noticed there was some confusion about keyword coloring.
![hedy](https://user-images.githubusercontent.com/1009735/151810764-d7efb3ec-2487-483b-b1b9-41c2faa5867d.png)
At what moment during the editing of text should the recognition of a keyword happen? There might be room for improvement here

**Comment by OnnoEbbens** [comment]
There is an open issue with highlighting echo, see #724

**Reply by StephanEggermont** [reply]
That sounds like a different issue. A solution for both at the same time would be nice of course

**Reply by OnnoEbbens** [reply]
Ah sorry I see it is not the same issue now. It sure would be nice if both can be fixed.
